src/app/app.ts

```diff
@@ -6,28 +6,42 @@ import {ENV, errorWithMessage} from '../utils';
 
 import {mattermost, zendesk} from '../clients';
 
-import store from '../store/config';
+import {config, oauth} from '../store';
 
 import {getTicketForPost} from './model';
 
 class App {
-    createTicketFromPost = async (call: AppCall): Promise<string> => {
+    createTicketFromPost = async (call: AppCall): Promise<void> => {
+        // get user ID for active Mattermost user
+        const userID = call.context.acting_user_id || '';
+        const [token, found] = oauth.getToken(userID);
+        if (found === false) {
+            throw new Error('Failed to get user access_token');
+        }
+
+        // get zendesk client for user
+        const zdClient = zendesk.newClient(ENV.zendesk.apiURL, token);
+
+        // create the ticket object from the form response
         const ticket = getTicketForPost(call.values);
-        const zdClient = zendesk.newClient(ENV.zendesk.username, ENV.zendesk.apiToken, ENV.zendesk.apiURL);
+
+        // create the ticket in Zendesk
         let result: Promise<ResponsePayload>;
         try {
             result = await zdClient.tickets.create(ticket);
         } catch (err) {
             throw new Error(errorWithMessage(err, 'Failed to create ticket'));
         }
 
+        // get the Zendesk user
         let user: Promise<ResponsePayload>;
```

Comment:
maybe denote is as smth. like zduser to not confuse it with the userId above.

Reply:
done. I also made a few other naming changes